backup/src/jsMain/kotlin/com/wire/backup/data/BackupDateTime.kt

```diff
@@ -0,0 +1,44 @@
+/*
+ * Wire
+ * Copyright (C) 2024 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ */
+package com.wire.backup.data
+
+import kotlin.js.Date
+
+@JsExport
+actual data class BackupDateTime(val date: Date) {
+    override fun equals(other: Any?): Boolean {
+        if (this === other) return true
+        if (other == null || this::class.js != other::class.js) return false
+
+        other as BackupDateTime
+
+        return this.toLongMilliseconds() == other.toLongMilliseconds()
+    }
```

Comment:
For some reason `backupDateTime == backupDateTime` doesn't work on JS, so I had to do it myself like so.